test.cpp

```diff
@@ -58,6 +58,28 @@ int main(int argc, char **argv) {
         std::cout << "    - " << k.dump() << "\n";
     }
 
+#ifdef JSON11_COMMENTS
```

Comment:
I'd like to see a few more test cases to test some of the tricky cases, such as intermixing comments and whitespace of both types to make sure the "what to do next" logic is correct.  Also tests for the ability to embed comments of one type within the other without interference.

Please also add some negative tests to ensure proper errors are reported by things like unterminated multiline comments